gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/DefaultBigQueryRpc.java

```diff
@@ -417,4 +433,69 @@ public QueryResponse query(QueryRequest request) throws BigQueryException {
       throw translate(ex);
     }
   }
+
+  @Override
+  public String open(JobConfigurationLoad configuration) throws BigQueryException {
+    try {
+      Job loadJob = new Job().setConfiguration(new JobConfiguration().setLoad(configuration));
+      StringBuilder builder = new StringBuilder()
+          .append(BASE_RESUMABLE_URI)
+          .append(options.projectId())
+          .append("/jobs");
+      GenericUrl url = new GenericUrl(builder.toString());
+      url.set("uploadType", "resumable");
+      JsonFactory jsonFactory = bigquery.getJsonFactory();
+      HttpRequestFactory requestFactory = bigquery.getRequestFactory();
+      HttpRequest httpRequest =
+          requestFactory.buildPostRequest(url, new JsonHttpContent(jsonFactory, loadJob));
+      httpRequest.getHeaders().set("X-Upload-Content-Value", "application/octet-stream");
+      HttpResponse response = httpRequest.execute();
+      return response.getHeaders().getLocation();
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public void write(String uploadId, byte[] toWrite, int toWriteOffset, long destOffset, int length,
```

Comment:
Is it possible to overwrite bytes with a request?  I would guess yes, but I figured I'd ask anyways just because I didn't see that in any example on the resumable upload docs.